src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.MultiAgent.cs

```diff
@@ -833,6 +839,39 @@ private static ulong CurlSendCallback(IntPtr buffer, ulong size, ulong nitems, I
                 return Interop.Http.CURL_READFUNC_ABORT;
             }
 
+            private static void SetLowSpeedLimit(EasyRequest easy, TimeSpan timeout)
+            {
+                // If there's no send/receive timeout on the handler, the defaults will
+                // remain disabled, and we don't need to worry about setting or resetting.
+                if (!easy._handler._hasSendReceiveTimeout)
+                {
+                    return;
+                }
+
+                // We need to ensure the timeout is set appropriately, which may include
+                // disabling a timeout currently set if one was.
+
+                long limit = 0, time = 0; // 0 == disable
+
+                if (timeout != Timeout.InfiniteTimeSpan)
+                {
+                    double secs = timeout.TotalSeconds;
+                    if (secs >= 1)
+                    {
+                        limit = 1;
```

Comment:
The origin of these timeouts came from the native WinHTTP API.  And that API (Win32 API) didn't have other notions of timeouts like CancellationToken, etc.  If you didn't use those timeouts, then the only recourse was to close the handle and abort everything.

Based on this conversation, we should probably pause and re-consider some of this.  

We do know that there are particular scenarios we need to solve. Sometimes you don't know the overall response body size (it will come down as chunked), so you don't know how long to wait overall before you give up. But you do want to have a timeout per "read" of each piece of data. If the server goes away suddenly or the network goes down, the socket read will hang.  Clearly, we need to make sure that "timeouts" (CancellationToken) on the stream.Read* of the response body and/or request body work as intended. I think that currently functionality is broken right now.

There are other scenarios where you want an "infinite" read timeout.  HTTP long polling scenario says that data might only arrive whenever the server feels like sending something new (i.e. stocker ticker scenario). So, thus you need to be able to deal with long timeouts but also be able to just cancel the whole HTTP operation and start over.